sans/convert_helpers.py

```diff
@@ -0,0 +1,301 @@
+        self.notify()
+
+    def _on_input_changed(self, change_dict):
+        if change_dict['name'] == 'value':
+            try:
+                input = self.scope[change_dict['new']]
+                self.convert_from.options = [
+                    key for key in allowed_conversions.keys()
+                    if key in input.coords
+                ]
+            except KeyError:
+                print(f'{change_dict["owner"].value} does not exist.')
+            self.convert_from.disabled = False
+            self.update_output()
+
+    def _on_from_changed(self, change_dict):
+        if change_dict['name'] == 'value':
+            allowed_dimensions = change_dict['owner'].options if change_dict[
+                'owner'].options else allowed_conversions.keys()
+            if change_dict['new'] not in allowed_dimensions:
+                print(
+                    f"{change_dict['new']} not a recognised conversion dimension. Dimensions in data are {allowed_dimensions}"
+                )
+                return
+            self.convert_to.options = allowed_conversions[change_dict['new']]
+            self.update_output()
+
+    def _on_to_changed(self, change_dict):
+        if change_dict['name'] == 'value':
+            allowed_dimensions = change_dict['owner'].options if change_dict[
+                'owner'].options else allowed_conversions.keys()
+            if change_dict['new'] not in allowed_dimensions:
+                print(
+                    f"{change_dict['new']} not a recognised conversion dimension. Dimensions supported are {allowed_dimensions}"
+                )
+                return
+            self.update_output()
+
+    def update_output(self, changes=None):
+        if not self.output.value and self.convert_to.value and self.input.value:
+            self.output.value = self.input.value + '_' + self.convert_to.value
+
+
+class PlotWidget(w.Box):
+    def __init__(self, scope):
+        super().__init__()
+        self.scope = scope
+        options = [
+            key for key, item in globals().items()
+            if isinstance(item, (sc.DataArray, sc.Dataset))
+        ]
+        self._data_selector = w.Combobox(placeholder='Data to plot',
+                                         options=options)
+        self._button = w.Button(description='Plot')
+        self._button.on_click(self._on_button_clicked)
+        self.plot_options = w.Output()  #HTML()
+        self.update_button = w.Button(description='Manual Update')
+        self.update_button.on_click(self.update)
+        self.output = w.Output(width='100%', height='100%')
+        self.children = [
+            w.VBox([
+                w.HBox([self.plot_options, self.update_button]),
+                w.HBox([self._data_selector, self._button]), self.output
+            ])
+        ]
+        self.update()
+
+    def _on_button_clicked(self, b):
+        self.output.clear_output()
+        with self.output:
```

Comment:
All this catching of matplotlib output should become much easier once the new plotting has been merged, since the figure will now itself be a widget, that can be placed in any widget box.